sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationExec.scala

```diff
@@ -274,6 +275,14 @@ object HiveScriptIOSchema extends HiveInspectors {
     var propsMap = serdeProps.toMap + (serdeConstants.LIST_COLUMNS -> columns.mkString(","))
     propsMap = propsMap + (serdeConstants.LIST_COLUMN_TYPES -> columnTypesNames)
 
+    if (!propsMap.contains(serdeConstants.SERIALIZATION_LAST_COLUMN_TAKES_REST)) {
+      val schemaProps = AnnotationUtils.getAnnotation(serde.getClass, classOf[SerDeSpec])
+      if (schemaProps != null &&
+        schemaProps.schemaProps().contains(serdeConstants.SERIALIZATION_LAST_COLUMN_TAKES_REST)) {
+        propsMap = propsMap + (serdeConstants.SERIALIZATION_LAST_COLUMN_TAKES_REST -> "true")
+      }
+    }
```

Comment:
This code to handle the schemaless case is the same with hive?

Reply:
> This code to handle the schemaless case is the same with hive?

No， since when hive plan query, it will set this config.
I think to keep same behavior with hive is ok

Reply:
> This code to handle the schemaless case is the same with hive?

No, since hive will set serde propertie when build plan,  but keep same behavior